doc/advanced-configuration/listen.md

```diff
@@ -0,0 +1,738 @@
+* **Default:** not set
+* **Example:** `tls.mode = "starttls"`
+
+By default there is no encryption for the incoming connections. You can change this by setting the `tls.mode` option to one of the following modes:
+
+* `tls` - clients must initiate a TLS session immediately after connecting, before beginning the normal XML stream,
+* `starttls` - enables StartTLS support; requires `certfile`,
+* `starttls_required` - enables and enforces StartTLS usage.
+
+#### `listen.c2s.tls.verify_peer`
+* **Syntax:** boolean
+* **Default:** `false`
+* **Example:** `verify_peer = true`
+
+Enforces verification of a client certificate. Requires a valid `cacertfile`.
+
+#### `listen.c2s.tls.module`
+* **Syntax:** string, one of `"just_tls"`, `"fast_tls"`
+* **Default:** `"fast_tls"`
+* **Example:** `tls.module = "just_tls"`
+
+By default the TLS library used for C2S connections is `fast_tls`, which uses OpenSSL-based NIFs. It is possible to change it to `just_tls` - Erlang TLS implementation provided by OTP. Some TLS-related options described here have different formats for these two libraries.
+
+#### `listen.c2s.tls.verify_mode` - for `just_tls`
+* **Syntax:** string, one of `"peer"`, `"selfsigned_peer"`, `"none"`
+* **Default:** not set (equivalent to `"peer"` in the current version of Erlang/OTP)
+* **Example:** `tls.verify_mode = "selfsigned_peer"`
+
+Specifies the way certificate verification works:
+
+* `peer` - makes sure the peer's certificate is valid and signed by a trusted CA,
+* `selfsigned_peer` - makes sure the peer's certificate is valid, but allows self-signed certificates,
+* `none` - any certificate is accepted.
+
+#### `listen.c2s.tls.disconnect_on_failure` - for `just_tls`
+* **Syntax:** boolean
+* **Default:** `true`
+* **Example:** `tls.disconnect_on_failure = false`
+
+Requires setting `tls.verify_mode`.  When set to `false`, it allows the client to connect even though the certificate verification failed. It is then up to the authentication layer to accept or reject the client connection. This behaviour mimcs the FastTLS one.
+
+#### `listen.c2s.tls.certfile`
+* **Syntax:** string, path in the file system
+* **Default:** not set
+* **Example:** `tls.certfile = "server.pem"`
+
+Path to the X509 PEM file with a certificate and a private key (not protected by a password). If the certificate is signed by an intermediate CA, you should specify here the whole CA chain by concatenating all public keys together and appending the private key after that.
+
+#### `listen.c2s.tls.cacertfile`
+* **Syntax:** string, path in the file system
+* **Default:** not set
+* **Example:** `tls.cacertfile = "ca.pem"`
+
+Path to the X509 PEM file with a CA chain that will be used to verify clients. Won't have any effect if `verify_peer` is not enabled.
+
+#### `listen.c2s.tls.dhfile`
+* **Syntax:** string, path in the file system
+* **Default:** not set
+* **Example:** `tls.dhfile = "dh.pem"`
+
+Path to the Diffie Hellman parameter file.
```

Comment:
It's not immediately obvious where these files should be put, maybe mention absolute and relative possibilities here?

Reply:
I am not sure if this is the place... the old docs did not say so and I wanted to avoid redundancy and repeating the same over and over again.